### PR TITLE
Added support @DisplayName annotation

### PR DESCRIPTION
**Pull request by nick130589**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 1]
```diff
@@ -357,7 +357,8 @@ public void testDisabled(ExtensionContext context, Optional<String> reason) {
 				String rawDescription = createStepDescription(context, itemType);
 				return StringUtils.isNotBlank(rawDescription) ? MarkdownUtils.asTwoParts(r, rawDescription) : r;
 			}).orElse(null);
-			startTestItem(context, Collections.emptyList(), itemType, description, null);
+			String methodName = reason.orElse(createStepMethodName(context, itemType));
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java:360]: I can't recall, that I asked to change this? What's this for?

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 2]
```diff
@@ -786,7 +788,25 @@ protected FinishTestItemRQ buildFinishTestItemRq(@Nonnull ExtensionContext conte
 	 */
 	protected String createStepName(ExtensionContext context) {
 		String name = context.getDisplayName();
-		return name.length() > 1024 ? name.substring(0, 1021) + "..." : name;
+		String defaultValue = name.length() > 1024 ? name.substring(0, 1021) + "..." : name;
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java:791]: The same logic is repeated 3 time in one method, please consider refactoring or moving it into separate method.

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 3]
```diff
@@ -849,6 +869,33 @@ protected String createStepDescription(ExtensionContext context, final ItemType
 		return defaultValue;
 	}
 
+	protected String createStepMethodName(ExtensionContext context, final ItemType itemType) {
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java:872]: Why do you need absolutely the same method twice?

**Review by HardNorth — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 1 above)
Comment (nick130589) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Removed it.

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 3 above)
Comment (nick130589) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Made one method.Resolved

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 2 above)
Comment (nick130589) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Extract to private method this logic

**Comment by nick130589** [comment]
All comments resolved

**Review by HardNorth — APPROVED** [review]

**Pull request merged**